src/compiler/parser.ts

```diff
@@ -6015,6 +6026,14 @@ namespace ts {
             return nextToken() === SyntaxKind.SlashToken;
         }
 
+        function isStartOfSimpleArrowFunction () {
+            if (token() === SyntaxKind.AsyncKeyword) {
+                nextToken();
+                return isIdentifier() && nextToken() === SyntaxKind.EqualsGreaterThanToken || token() === SyntaxKind.EqualsGreaterThanToken;

```

Comment:
Can you leave a comment on the intent here? Is this supposed to cover `async => 100`?